generate_index.py

```diff
@@ -1,30 +1,116 @@
+from __future__ import annotations
+
+import argparse
+from collections import defaultdict
+import json
 import os
+from pathlib import Path
+import sys
+from packaging.utils import parse_wheel_filename
+from urllib.parse import urlparse
+from textwrap import dedent
 
 import github
 
+##
+## Output a PEP 503 compliant package repository for Pants wheels.
+## See https://peps.python.org/pep-0503/
+##
+
 
-def main() -> str:
-    gh = github.Github(auth=github.Auth.Token(os.environ["GH_TOKEN"]))
+def get_pants_python_packages(gh: github.Github) -> tuple[str, ...]:
     repo = gh.get_repo("pantsbuild/pants")
-    releases = repo.get_releases()
-    index = "\n".join(
-        [
-            "<html>",
-            "<body>",
-            "<h1>Links for Pantsbuild Wheels</h1>",
-            *(
-                f'<a href="{asset.browser_download_url}">{asset.name}</a><br>'
-                for release in releases
-                if release.tag_name.startswith("release_2")
-                for asset in release.assets
-                if asset.name.endswith(".whl")
-            ),
-            "</body>",
-            "</html>",
-        ]
-    )
-    return index
+    all_releases = repo.get_releases()
+
+    pants_wheel_assets = [
+        asset
+        for release in all_releases
+        if release.tag_name.startswith("release_2")
+        for asset in release.assets
+        if asset.name.endswith(".whl")
+    ]
+
+    packages = defaultdict(lambda: defaultdict(list))
+
+    for asset in pants_wheel_assets:
+        name, version, build_tag, tags = parse_wheel_filename(asset.name)
+        packages[name][version].append(asset)
+    
+    return packages
+
+
+def main(args):
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--url-path-prefix", default="/", action="store")
```

Comment:
My read of pep-0503 is that the URLs must always be exactly `/<project>/`, i.e. customising this `--url-path-prefix` value may not work at all, and so potentially shouldn't be exposed?

Reply:
My read of PEP 503 is that they are the full path with trailing `/`. And checking PyPi itself shows that the links include the prefix since they are `/simple/PKG_NAME/`. (For example, look at `curl https://pypi.org/simple/ -o pypi.html`.)